src/serde/mod.rs

```diff
@@ -0,0 +1,271 @@
+//!
+//! * `to_string`: Serialize an object into compact Ion text format.
+//! * `to_pretty`: Serialize an object into pretty formatted Ion text.
+//! * `to_binary`:  Serialize an object into Ion binary format.
+//!
+//! For deserialization `from_ion` API is provided through this module.
+//!
+//! ## Mapping of Ion data types to Rust and serde data types
+//!
+//!| Ion data type | Rust data structure | Serde data type |
+//!|---------------|---------------------|-----------------|
+//!| int           | u64, i64, u32, i32, u16, i16, u8, i8 | u64, i64, u32, i32, u16, i16, u8, i8 |
+//!| float         | f32, f64            | f32, f64  |
+//!| decimal       | Decimal(Ion Element API) | newtype_struct (with name as `$__ion_rs_decimal__`) |
+//!| timestamp       | Timestamp(Ion Element API) | newtype_struct (with name as `$__ion_rs_timestamp__`) |
+//!| blob          | byte array               | byte array |
+//!| clob          | byte array               | byte array |
+//!| bool          | bool                | bool |
+//!| symbol        | string              | string |
+//!| string        | string              | string |
+//!| struct        | struct              | struct |
+//!| list          | vector              | seq |
+//!| null          | None                | unit |
+//!
+//! ## Mapping of serde data types to Ion data types
+//!
+//!| Serde data type | Ion data type |
+//!|---------------|---------------------|
+//!| u64, i64, u32, i32, u16, i16, u8, i8 | int |
+//!| char, string, unit_variant | string |
+//!| byte-array | blob |
+//!| option | None - null, Some - based on other mappings |
+//!| unit, unit_struct | null |
+//!| seq, tuple, tuple_struct, tuple_variant | list |
+//!| newtype_struct ,newtype_variant, map, struct, struct_variant | struct |
+//!
+//! _Note: Since the serde framework doesn't support [Ion decimal] and [Ion timestamp] types, distinct serialization and deserialization of these types are defined in this module.
+//! It uses `newtype_struct` with `$__ion_rs_decimal__` and `$__ion_rs_timestamp__` as struct names from [serde data model],
+//! to indicate serde framework to use Ion's implementation of decimal and timestamp serialization and deserialization.
+//! If one wants to use [chrono::DateTime], it needs to be tagged with `#[serde_as(as = crate::Timestamp)]`._
+//!
+//! ## Example of serialization of Rust struct into Ion data
+//! ```
+//! use ion_rs::IonResult;
+//! use crate::ion_rs::serde::to_string;
+//! use serde::{Deserialize, Serialize};
+//!
+//!#[derive(Serialize, Deserialize)]
+//! struct Address {
+//!     street: String,
+//!     city: String,
+//! }
+//!
+//! fn main() -> IonResult<()> {
+//!     // data structure for representing address
+//!     let address = Address {
+//!         street: "10 Downing Street".to_owned(),
+//!         city: "London".to_owned(),
+//!     };
+//!
+//!     // serialize it to an Ion data
+//!     let ion = to_string(&address)?;
```

Comment:
If it's meant to be a free-standing function, how about a name like `to_ion_text()`? Is there a way to pass in a writer of your own so you can configure it for `pretty`/`lines`/whatever?

Reply:
currently we have different methods for to_string, to_pretty, ot_binary. but yes, I can modify it to be a single method with configuration to the writer.

Reply:
https://github.com/amazon-ion/ion-rust/issues/646